alg.go

```diff
@@ -75,7 +75,7 @@ func (l *limiter) pschedule() {
 			return
 		default:
 			l.ltask.Lock()
-			if len(l.ltask.tasks) > 0 {
+			if len(l.ltask.tasks) > 1 {
```

Comment:
if the slice size equals to 1, the next line will trigger `panic: runtime error: index out of range`